src/api/auth.ts

```diff
@@ -33,3 +34,12 @@ export function useSignUp() {
     },
   });
 }
+
+export function useSignOut() {
+  return useMutation<void, FirebaseError, void>({
+    mutationFn: async () => {
+      const responseSingOut = await signOut(auth);
+      return responseSingOut;
```

Comment:
response of SignOut is of type void so I don't think you need to return anything, you can also remove `const responseSingOut` and just use `await signOut(auth)`